drivers/serial/uart_xlnx_ps.c

```diff
@@ -313,6 +385,446 @@ static void uart_xlnx_ps_poll_out(struct device *dev, unsigned char c)
+ *
+ * Converts a bit mask representing the UART's parity setting within
+ * the UART's Mode Register into a value of an enumeration type provided
+ * by the UART driver API.
+ *
+ * @param mode_reg The current Mode Register contents from which the
+ *                 parity setting shall be extracted.
+ *
+ * @return The current parity setting mapped to the UART driver API's
+ *         enum type.
+ */
+static inline enum uart_config_parity uart_xlnx_ps_ll2cfg_parity(
+	u32_t mode_reg)
+{
+	/*
+	 * Obtain the current parity configuration from the mode register's
+	 * bits [5..3] (PAR):
+	 *  000b : even -> reset value
+	 *  001b : odd
+	 *  010b : space
+	 *  011b : mark
+	 *  1xxb : none
+	 */
+
+	switch ((mode_reg & XUARTPS_MR_PARITY_MASK)) {
+	case XUARTPS_MR_PARITY_EVEN:
+	default:
+		return UART_CFG_PARITY_EVEN;
+	case XUARTPS_MR_PARITY_ODD:
+		return UART_CFG_PARITY_ODD;
+	case XUARTPS_MR_PARITY_SPACE:
+		return UART_CFG_PARITY_SPACE;
+	case XUARTPS_MR_PARITY_MARK:
+		return UART_CFG_PARITY_MARK;
+	case XUARTPS_MR_PARITY_NONE:
+		return UART_CFG_PARITY_NONE;
+	}
+}
+
+/**
+ * @brief Converts a Mode Register bit mask to a stop bit configuration
+ *        enum value.
+ *
+ * Converts a bit mask representing the UART's stop bit setting within
+ * the UART's Mode Register into a value of an enumeration type provided
+ * by the UART driver API.
+ *
+ * @param mode_reg The current Mode Register contents from which the
+ *                 stop bit setting shall be extracted.
+ *
+ * @return The current stop bit setting mapped to the UART driver API's
+ *         enum type.
+ */
+static inline enum uart_config_stop_bits uart_xlnx_ps_ll2cfg_stopbits(
+	u32_t mode_reg)
+{
+	/*
+	 * Obtain the current stop bit configuration from the mode register's
+	 * bits [7..6] (NBSTOP):
+	 *  00b : 1 stop bit -> reset value
+	 *  01b : 1.5 stop bits
+	 *  10b : 2 stop bits
+	 *  11b : reserved
+	 */
+
+	switch ((mode_reg & XUARTPS_MR_STOPMODE_MASK)) {
+	case XUARTPS_MR_STOPMODE_1_BIT:
+	default:
+		return UART_CFG_STOP_BITS_1;
+	case XUARTPS_MR_STOPMODE_1_5_BIT:
+		return UART_CFG_STOP_BITS_1_5;
+	case XUARTPS_MR_STOPMODE_2_BIT:
+		return UART_CFG_STOP_BITS_2;
+	}
+}
+
+/**
+ * @brief Converts a Mode Register bit mask to a data bit configuration
+ *        enum value.
+ *
+ * Converts a bit mask representing the UART's data bit setting within
+ * the UART's Mode Register into a value of an enumeration type provided
+ * by the UART driver API.
+ *
+ * @param mode_reg The current Mode Register contents from which the
+ *                 data bit setting shall be extracted.
+ *
+ * @return The current data bit setting mapped to the UART driver API's
+ *         enum type.
+ */
+static inline enum uart_config_data_bits uart_xlnx_ps_ll2cfg_databits(
+	u32_t mode_reg)
+{
+	/*
+	 * Obtain the current data bit configuration from the mode register's
+	 * bits [2..1] (CHRL):
+	 *  0xb : 8 data bits -> reset value
+	 *  10b : 7 data bits
+	 *  11b : 6 data bits
+	 */
+
```

Comment:
Extra empty line, would prefer no empty lines between the comments and the code.